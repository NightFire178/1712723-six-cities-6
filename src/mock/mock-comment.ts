import mockUser from "./mock-user";
import comment from '../types/comment'

const mockComment:comment = {
  comment: `Beautiful space, fantastic location and atmosphere,
  really a wonderful place to spend a few days. Will be back.`,
  date: `2021-03-07T08:04:28.647Z`,
  id: 10,
  rating: 4,
  user: mockUser
}
export default mockComment

export const mockArrayComments:Array<comment> =[
  {
    comment: `Beautiful space, fantastic location and atmosphere,
  really a wonderful place to spend a few days. Will be back.`,
    date: `2021-03-07T08:04:28.647Z`,
    id: 15,
    rating: 3,
    user: mockUser
  },
  {
    comment: `Beautiful space, fantastic location and atmosphere,
  really a wonderful place to spend a few days. Will be back.`,
    date: `2021-03-07T08:04:28.647Z`,
    id: 13,
    rating: 5,
    user: mockUser
  },
  {
    comment: `What an amazing view! The house is stunning and in
    an amazing location. The large glass wall had an amazing view of the river!`,
    date: `2021-03-07T08:04:28.647Z`,
    id: 10,
    rating: 4,
    user: mockUser
  }
]
