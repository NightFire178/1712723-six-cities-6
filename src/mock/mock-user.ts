import User from "../types/user";

const mockUser:User = {
  id: 1,
  name: `vasa`,
  is_pro: true,
  avatar_url:`https://assets.htmlacademy.ruâ€¦tar/6.jp`,
  email: `sdfhskd@maid.ru`
}
export default mockUser
