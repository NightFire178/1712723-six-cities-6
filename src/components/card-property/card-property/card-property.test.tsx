import React from 'react'
import {render} from '@testing-library/react'
import CardProperty from './card-property'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import mockOneHotel, {mockArrayHotels} from "../../../mock/mock-hotels";
import {mockArrayComments} from "../../../mock/mock-comment";
import mockUser from "../../../mock/mock-user";

const useHotelsSelection = {
  oneHotel() {
    return mockOneHotel
  }
}

jest.mock("../../../hooks/use-selectors-state/use-hotels", ()=>useHotelsSelection)
const  useHotelInfoSelection = {
  hotelInfoNearby()  {
    return [mockArrayHotels[0], mockArrayHotels[1], mockArrayHotels[2]]
  },
  hotelInfoComment() {
    return mockArrayComments
  }
}
jest.mock("../../../hooks/use-selectors-state/use-hotel-info", ()=>useHotelInfoSelection)
const useAppStateSelection = {
  isAuth() {
    return {
      now:true,
      user:mockUser
    }
  }
}
const thunkHotelInfo =()=> new Promise((resolve)=>{resolve(200)})
jest.mock('../../../hooks/use-thunk',() =>()=>({thunkHotelInfo}))
jest.mock("../../../hooks/use-selectors-state/use-app-state", ()=>useAppStateSelection)
// eslint-disable-next-line
jest.mock("../../block/favorite-button/favorite-button", ()=>()=><div>FavoriteButton</div>)
// eslint-disable-next-line
jest.mock("../comments/comments/comments", ()=>()=><div>Comments</div>)
// eslint-disable-next-line
jest.mock("../../block/card/card", ()=>()=><div>Card</div>)
// eslint-disable-next-line
jest.mock("../../block/header/header", ()=>()=><div>Header</div>)
test(`card-property component test`, () => {
  window.scrollTo = jest.fn()
  const history = createMemoryHistory()
  const match = {params: {id: `2`}}
  const city = render(
    <Router history={history}>
      <CardProperty match={match}/>
    </Router>
  )
  expect(city).toMatchSnapshot()
})
