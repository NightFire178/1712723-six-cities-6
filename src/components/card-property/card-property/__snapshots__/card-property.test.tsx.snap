// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card-property component test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          style="display: none;"
        >
          <svg
            xmlns="http://www.w3.org/2000/svg"
          >
            <symbol
              id="icon-arrow-select"
              viewBox="0 0 7 4"
            >
              <path
                clip-rule="evened"
                d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
                fill-rule="evenodd"
              />
            </symbol>
            <symbol
              id="icon-bookmark"
              viewBox="0 0 17 18"
            >
              <path
                d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
              />
            </symbol>
            <symbol
              id="icon-star"
              viewBox="0 0 13 12"
            >
              <path
                clip-rule="evened"
                d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
                fill-rule="evenodd"
              />
            </symbol>
          </svg>
        </div>
        <div
          class="page"
        >
          <div>
            Header
          </div>
          <main
            class="page__main page__main--property"
          >
            <section
              class="property"
            >
              <div
                class="property__gallery-container container"
              >
                <div
                  class="property__gallery"
                >
                  <div
                    class="property__image-wrapper"
                  >
                    <img
                      alt="Photo studio"
                      class="property__image"
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg"
                    />
                  </div>
                  <div
                    class="property__image-wrapper"
                  >
                    <img
                      alt="Photo studio"
                      class="property__image"
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg"
                    />
                  </div>
                  <div
                    class="property__image-wrapper"
                  >
                    <img
                      alt="Photo studio"
                      class="property__image"
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg"
                    />
                  </div>
                  <div
                    class="property__image-wrapper"
                  >
                    <img
                      alt="Photo studio"
                      class="property__image"
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg"
                    />
                  </div>
                  <div
                    class="property__image-wrapper"
                  >
                    <img
                      alt="Photo studio"
                      class="property__image"
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
                    />
                  </div>
                  <div
                    class="property__image-wrapper"
                  >
                    <img
                      alt="Photo studio"
                      class="property__image"
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg"
                    />
                  </div>
                </div>
              </div>
              <div
                class="property__container container"
              >
                <div
                  class="property__wrapper"
                >
                  <div
                    class="property__mark"
                  >
                    <span>
                      Premium
                    </span>
                  </div>
                  <div
                    class="property__name-wrapper"
                  >
                    <h1
                      class="property__name"
                    >
                      The house among olive 
                    </h1>
                    <div>
                      FavoriteButton
                    </div>
                  </div>
                  <div
                    class="property__rating rating"
                  >
                    <div
                      class="property__stars rating__stars"
                    >
                      <span
                        style="width: 60%;"
                      />
                      <span
                        class="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                    <span
                      class="property__rating-value rating__value"
                    >
                      3
                    </span>
                  </div>
                  <ul
                    class="property__features"
                  >
                    <li
                      class="property__feature property__feature--entire"
                    >
                      room
                    </li>
                    <li
                      class="property__feature property__feature--bedrooms"
                    >
                      1
                       Bedrooms
                    </li>
                    <li
                      class="property__feature property__feature--adults"
                    >
                      Max 
                      1
                       adults
                    </li>
                  </ul>
                  <div
                    class="property__price"
                  >
                    <b
                      class="property__price-value"
                    >
                      €
                      169
                    </b>
                    <span
                      class="property__price-text"
                    >
                       night
                    </span>
                  </div>
                  <div
                    class="property__inside"
                  >
                    <h2
                      class="property__inside-title"
                    >
                      What\`s inside
                    </h2>
                    <ul
                      class="property__inside-list"
                    >
                      <li
                        class="property__inside-item"
                      >
                        Breakfast
                      </li>
                      <li
                        class="property__inside-item"
                      >
                        Air conditioning
                      </li>
                      <li
                        class="property__inside-item"
                      >
                        Laptop friendly workspace
                      </li>
                      <li
                        class="property__inside-item"
                      >
                        Washer
                      </li>
                    </ul>
                  </div>
                  <div
                    class="property__host"
                  >
                    <h2
                      class="property__host-title"
                    >
                      Meet the host
                    </h2>
                    <div
                      class="property__host-user user"
                    >
                      <div
                        class="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
                      >
                        <img
                          alt="Host avatar"
                          class="property__avatar user__avatar"
                          height="74"
                          src="img/avatar-angelina.jpg"
                          width="74"
                        />
                      </div>
                      <span
                        class="property__user-name"
                      >
                        Angelina
                      </span>
                    </div>
                    <div
                      class="property__description"
                    >
                      <p
                        class="property__text"
                      >
                        Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country
                        .
                      </p>
                      <p
                        class="property__text"
                      >
                         In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed
                        .
                      </p>
                      <p
                        class="property__text"
                      >
                        
                        .
                      </p>
                    </div>
                  </div>
                  <div>
                    Comments
                  </div>
                </div>
              </div>
              <section
                class="property__map map"
                style="width: 1144px; margin: 0px auto;"
              >
                <div
                  class="leaflet-container leaflet-grab leaflet-touch-drag"
                  id="map"
                  style="height: 100%; position: relative;"
                  tabindex="0"
                >
                  <div
                    class="leaflet-pane leaflet-map-pane"
                    style="left: 0px; top: 0px;"
                  >
                    <div
                      class="leaflet-pane leaflet-tile-pane"
                    >
                      <div
                        class="leaflet-layer "
                        style="z-index: 1;"
                      >
                        <div
                          class="leaflet-tile-container leaflet-zoom-animated"
                          style="z-index: 18; left: 0px; top: 0px;"
                        >
                          <img
                            alt=""
                            class="leaflet-tile"
                            role="presentation"
                            src="https://b.basemaps.cartocdn.com/rastertiles/voyager/13/4195/2748.png"
                            style="width: 256px; height: 256px; left: -6px; top: -9px;"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class="leaflet-pane leaflet-shadow-pane"
                    />
                    <div
                      class="leaflet-pane leaflet-overlay-pane"
                    />
                    <div
                      class="leaflet-pane leaflet-marker-pane"
                    >
                      <img
                        alt=""
                        class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                        src="img/pin.svg"
                        style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -29px; top: 19px; z-index: 19;"
                        tabindex="0"
                      />
                      <img
                        alt=""
                        class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                        src="img/pin.svg"
                        style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 15206px; top: -958px; z-index: -958;"
                        tabindex="0"
                      />
                      <img
                        alt=""
                        class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                        src="img/pin.svg"
                        style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 134px; top: -27px; z-index: -27;"
                        tabindex="0"
                      />
                      <img
                        alt=""
                        class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                        src="img/pin-active.svg"
                        style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -11657px; top: 18142px; z-index: 18142;"
                        tabindex="0"
                      />
                    </div>
                    <div
                      class="leaflet-pane leaflet-tooltip-pane"
                    />
                    <div
                      class="leaflet-pane leaflet-popup-pane"
                    />
                  </div>
                  <div
                    class="leaflet-control-container"
                  >
                    <div
                      class="leaflet-top leaflet-left"
                    />
                    <div
                      class="leaflet-top leaflet-right"
                    />
                    <div
                      class="leaflet-bottom leaflet-left"
                    />
                    <div
                      class="leaflet-bottom leaflet-right"
                    >
                      <div
                        class="leaflet-control-attribution leaflet-control"
                      >
                        <a
                          href="https://leafletjs.com"
                          title="A JS library for interactive maps"
                        >
                          Leaflet
                        </a>
                         | © 
                        <a
                          href="https://www.openstreetmap.org/copyright"
                        >
                          OpenStreetMap
                        </a>
                         contributors © 
                        <a
                          href="https://carto.com/attributions"
                        >
                          CARTO
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </section>
            </section>
            <div
              class="container"
            >
              <section
                class="near-places places"
              >
                <h2
                  class="near-places__title"
                >
                  Other places in the neighbourhood
                </h2>
                <div
                  class="near-places__list places__list"
                >
                  <div>
                    Card
                  </div>
                  <div>
                    Card
                  </div>
                  <div>
                    Card
                  </div>
                </div>
              </section>
            </div>
          </main>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        style="display: none;"
      >
        <svg
          xmlns="http://www.w3.org/2000/svg"
        >
          <symbol
            id="icon-arrow-select"
            viewBox="0 0 7 4"
          >
            <path
              clip-rule="evened"
              d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
              fill-rule="evenodd"
            />
          </symbol>
          <symbol
            id="icon-bookmark"
            viewBox="0 0 17 18"
          >
            <path
              d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
            />
          </symbol>
          <symbol
            id="icon-star"
            viewBox="0 0 13 12"
          >
            <path
              clip-rule="evened"
              d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
              fill-rule="evenodd"
            />
          </symbol>
        </svg>
      </div>
      <div
        class="page"
      >
        <div>
          Header
        </div>
        <main
          class="page__main page__main--property"
        >
          <section
            class="property"
          >
            <div
              class="property__gallery-container container"
            >
              <div
                class="property__gallery"
              >
                <div
                  class="property__image-wrapper"
                >
                  <img
                    alt="Photo studio"
                    class="property__image"
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg"
                  />
                </div>
                <div
                  class="property__image-wrapper"
                >
                  <img
                    alt="Photo studio"
                    class="property__image"
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg"
                  />
                </div>
                <div
                  class="property__image-wrapper"
                >
                  <img
                    alt="Photo studio"
                    class="property__image"
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg"
                  />
                </div>
                <div
                  class="property__image-wrapper"
                >
                  <img
                    alt="Photo studio"
                    class="property__image"
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg"
                  />
                </div>
                <div
                  class="property__image-wrapper"
                >
                  <img
                    alt="Photo studio"
                    class="property__image"
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
                  />
                </div>
                <div
                  class="property__image-wrapper"
                >
                  <img
                    alt="Photo studio"
                    class="property__image"
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg"
                  />
                </div>
              </div>
            </div>
            <div
              class="property__container container"
            >
              <div
                class="property__wrapper"
              >
                <div
                  class="property__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  class="property__name-wrapper"
                >
                  <h1
                    class="property__name"
                  >
                    The house among olive 
                  </h1>
                  <div>
                    FavoriteButton
                  </div>
                </div>
                <div
                  class="property__rating rating"
                >
                  <div
                    class="property__stars rating__stars"
                  >
                    <span
                      style="width: 60%;"
                    />
                    <span
                      class="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                  <span
                    class="property__rating-value rating__value"
                  >
                    3
                  </span>
                </div>
                <ul
                  class="property__features"
                >
                  <li
                    class="property__feature property__feature--entire"
                  >
                    room
                  </li>
                  <li
                    class="property__feature property__feature--bedrooms"
                  >
                    1
                     Bedrooms
                  </li>
                  <li
                    class="property__feature property__feature--adults"
                  >
                    Max 
                    1
                     adults
                  </li>
                </ul>
                <div
                  class="property__price"
                >
                  <b
                    class="property__price-value"
                  >
                    €
                    169
                  </b>
                  <span
                    class="property__price-text"
                  >
                     night
                  </span>
                </div>
                <div
                  class="property__inside"
                >
                  <h2
                    class="property__inside-title"
                  >
                    What\`s inside
                  </h2>
                  <ul
                    class="property__inside-list"
                  >
                    <li
                      class="property__inside-item"
                    >
                      Breakfast
                    </li>
                    <li
                      class="property__inside-item"
                    >
                      Air conditioning
                    </li>
                    <li
                      class="property__inside-item"
                    >
                      Laptop friendly workspace
                    </li>
                    <li
                      class="property__inside-item"
                    >
                      Washer
                    </li>
                  </ul>
                </div>
                <div
                  class="property__host"
                >
                  <h2
                    class="property__host-title"
                  >
                    Meet the host
                  </h2>
                  <div
                    class="property__host-user user"
                  >
                    <div
                      class="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
                    >
                      <img
                        alt="Host avatar"
                        class="property__avatar user__avatar"
                        height="74"
                        src="img/avatar-angelina.jpg"
                        width="74"
                      />
                    </div>
                    <span
                      class="property__user-name"
                    >
                      Angelina
                    </span>
                  </div>
                  <div
                    class="property__description"
                  >
                    <p
                      class="property__text"
                    >
                      Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country
                      .
                    </p>
                    <p
                      class="property__text"
                    >
                       In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed
                      .
                    </p>
                    <p
                      class="property__text"
                    >
                      
                      .
                    </p>
                  </div>
                </div>
                <div>
                  Comments
                </div>
              </div>
            </div>
            <section
              class="property__map map"
              style="width: 1144px; margin: 0px auto;"
            >
              <div
                class="leaflet-container leaflet-grab leaflet-touch-drag"
                id="map"
                style="height: 100%; position: relative;"
                tabindex="0"
              >
                <div
                  class="leaflet-pane leaflet-map-pane"
                  style="left: 0px; top: 0px;"
                >
                  <div
                    class="leaflet-pane leaflet-tile-pane"
                  >
                    <div
                      class="leaflet-layer "
                      style="z-index: 1;"
                    >
                      <div
                        class="leaflet-tile-container leaflet-zoom-animated"
                        style="z-index: 18; left: 0px; top: 0px;"
                      >
                        <img
                          alt=""
                          class="leaflet-tile"
                          role="presentation"
                          src="https://b.basemaps.cartocdn.com/rastertiles/voyager/13/4195/2748.png"
                          style="width: 256px; height: 256px; left: -6px; top: -9px;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="leaflet-pane leaflet-shadow-pane"
                  />
                  <div
                    class="leaflet-pane leaflet-overlay-pane"
                  />
                  <div
                    class="leaflet-pane leaflet-marker-pane"
                  >
                    <img
                      alt=""
                      class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                      src="img/pin.svg"
                      style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -29px; top: 19px; z-index: 19;"
                      tabindex="0"
                    />
                    <img
                      alt=""
                      class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                      src="img/pin.svg"
                      style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 15206px; top: -958px; z-index: -958;"
                      tabindex="0"
                    />
                    <img
                      alt=""
                      class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                      src="img/pin.svg"
                      style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 134px; top: -27px; z-index: -27;"
                      tabindex="0"
                    />
                    <img
                      alt=""
                      class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                      src="img/pin-active.svg"
                      style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -11657px; top: 18142px; z-index: 18142;"
                      tabindex="0"
                    />
                  </div>
                  <div
                    class="leaflet-pane leaflet-tooltip-pane"
                  />
                  <div
                    class="leaflet-pane leaflet-popup-pane"
                  />
                </div>
                <div
                  class="leaflet-control-container"
                >
                  <div
                    class="leaflet-top leaflet-left"
                  />
                  <div
                    class="leaflet-top leaflet-right"
                  />
                  <div
                    class="leaflet-bottom leaflet-left"
                  />
                  <div
                    class="leaflet-bottom leaflet-right"
                  >
                    <div
                      class="leaflet-control-attribution leaflet-control"
                    >
                      <a
                        href="https://leafletjs.com"
                        title="A JS library for interactive maps"
                      >
                        Leaflet
                      </a>
                       | © 
                      <a
                        href="https://www.openstreetmap.org/copyright"
                      >
                        OpenStreetMap
                      </a>
                       contributors © 
                      <a
                        href="https://carto.com/attributions"
                      >
                        CARTO
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </section>
          </section>
          <div
            class="container"
          >
            <section
              class="near-places places"
            >
              <h2
                class="near-places__title"
              >
                Other places in the neighbourhood
              </h2>
              <div
                class="near-places__list places__list"
              >
                <div>
                  Card
                </div>
                <div>
                  Card
                </div>
                <div>
                  Card
                </div>
              </div>
            </section>
          </div>
        </main>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
